src/lumigo_opentelemetry/__init__.py

```diff
@@ -297,15 +299,68 @@ def generate_trace_id(self) -> int:
     return {"tracer_provider": tracer_provider, "logger_provider": logger_provider}
 
 
+def _flush_with_timeout(args: List[Any]) -> None:
+    """
+    Flush with dynamic timeout based on Lambda context remaining time.
+    Supports environment variable override via LUMIGO_FLUSH_TIMEOUT.
+
+    Args:
+        args: Function arguments where args[1] should be Lambda context if available
+    """
+    from opentelemetry import trace
+
+    timeout_ms = DEFAULT_TIMEOUT_MS
+    try:
+        custom_timeout_env = os.getenv("LUMIGO_FLUSH_TIMEOUT")
+        if custom_timeout_env:
+            custom_timeout_ms = int(custom_timeout_env)
+            if custom_timeout_ms > 0:
+                timeout_ms = custom_timeout_ms
+                logger.debug(
+                    f"Using custom flush timeout from LUMIGO_FLUSH_TIMEOUT: {timeout_ms}ms"
+                )
+            else:
+                logger.warning(
+                    f"Invalid LUMIGO_FLUSH_TIMEOUT value '{custom_timeout_env}' (must be > 0). {USING_DEFAULT_TIMEOUT_MESSAGE}"
+                )
+        else:
+            if len(args) >= 2 and hasattr(args[1], "get_remaining_time_in_millis"):
+                context = args[1]
+                remaining_time_ms = context.get_remaining_time_in_millis()
+
+                buffer_time_ms = remaining_time_ms * 0.10
+                timeout_seconds = (remaining_time_ms - buffer_time_ms) / 1000
+                timeout_ms = int(timeout_seconds * 1000)
+
```

Comment:
shouldn't there be a minimum? what if the lambda has 15 minutes left. the timeout shouldn't be that long